src/main/java/org/folio/rest/impl/PermsAPI.java

```diff
@@ -1997,29 +1895,32 @@ private boolean allowAccessByPermission(String permissions, String permissionNam
         rpbnRes -> {
           if (rpbnRes.failed()) {
             promise.fail(rpbnRes.cause());
+            return;
+          }
+          if (rpbnRes.result() == null) {
+            promise.complete(false);
           } else {
-            if (rpbnRes.result() == null) {
-              promise.complete(false);
-            } else {
-              Boolean dummy = rpbnRes.result().getDummy();
-              if (dummy != null && dummy) {
-                promise.complete(true);
-              } else {
-                promise.complete(false);
-              }
-            }
+            promise.complete(Boolean.TRUE.equals(rpbnRes.result().getDummy()));
           }
         });
     return promise.future().compose(next -> {
-      if (next) {
+      if (Boolean.TRUE.equals(next)) {
         return Future.succeededFuture(true);
       } else {
         return checkPermlistForDummy(permListCopy, vertxContext, tenantId);
       }
     });
   }
 
-  private Criteria getIdCriteria(String indexField, String operation, String value) {
+  private static Criterion getIdCriterion(String id) {
+    Criteria idCrit = new Criteria();
+    idCrit.addField(ID_FIELD);
```

Comment:
Add `idCrit.setJSONB(false)` to make use of the primary key index resulting in best performance:
https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/persist/Criteria/Criteria.java#L30

Reply:
Done and ID_FIELD must be unquoted.